### DISCUSSION TITLE
Cannot build IPA file

### DISCUSSION THREAD

**Discussion by InvalidPandaa** [SoCuul/SCInsta#39] [Support] [answered]
<img width="1352" alt="image" src="https://github.com/SoCuul/SCInsta/assets/117936522/972ef11d-882b-407b-83d7-277dd4178dbd">

Hello, can anyone help me? I'm getting this error while building the IPA file

**Comment by Cody47** [comment]
just manually copy the libbhFLEX.dylib file to the /.theos/obj/debug directory!

**Reply by InvalidPandaa** [reply]
I don't have this file in the SCInsta directory

**Comment by InvalidPandaa** [comment]
Here's the output of build.sh

Building BHInsta tweak for sideloading (as IPA)
==> Cleaning…
> Making all for tweak BHInsta…
==> Preprocessing src/Features/General/CopyDescription.x…
==> Preprocessing src/Features/General/NoRecentSearches.x…
==> Preprocessing src/Features/StoriesAndMessages/DisableStorySeen.x…
==> Preprocessing src/Features/General/HideTrendingSearches.x…
==> Preprocessing src/Features/StoriesAndMessages/SeenButtons.x…
==> Preprocessing src/Features/StoriesAndMessages/KeepDeletedMessages.x…
==> Preprocessing src/Tweak.x…
==> Preprocessing src/Features/General/HideExploreGrid.xm…
==> Preprocessing src/Features/Confirm/StickerInteractConfirm.x…
==> Preprocessing src/Features/Confirm/PostCommentConfirm.x…
==> Preprocessing src/Features/Confirm/CallConfirm.x…
==> Preprocessing src/Features/General/BHSettingsMenuEntry.x…
==> Preprocessing src/Features/Confirm/FollowConfirm.x…
==> Preprocessing src/Features/Confirm/DMAudioMsgConfirm.x…
==> Preprocessing src/Features/Feed/HideSuggestedReels.x…
==> Preprocessing src/Features/Feed/HideSuggestedForYou.x…
==> Preprocessing src/Features/Feed/HideThreads.x…
==> Preprocessing src/Features/Confirm/LikeConfirm.x…
==> Preprocessing src/Features/Save/ProfileImageSave.x…
==> Preprocessing src/Features/Feed/HideAds.x…
==> Preprocessing src/Features/Save/StoriesSave.x…
==> Preprocessing src/SideloadFix.x…
==> Preprocessing src/Features/Save/MediaSave.x…
==> Compiling modules/JGProgressHUD/JGProgressHUDErrorIndicatorView.m (arm64)…
==> Compiling src/Controllers/SecurityViewController.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUDAnimation.m (arm64)…
==> Compiling src/Download.m (arm64)…
==> Compiling src/Utils.m (arm64)…
==> Compiling src/Manager.m (arm64)…
==> Compiling src/Controllers/SettingsViewController.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUD.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUDIndeterminateIndicatorView.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUDIndicatorView.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUDImageIndicatorView.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUDFadeAnimation.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUDFadeZoomAnimation.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUDRingIndicatorView.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUDPieIndicatorView.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUDShadow.m (arm64)…
==> Compiling modules/JGProgressHUD/JGProgressHUDSuccessIndicatorView.m (arm64)…
==> Compiling src/Features/StoriesAndMessages/KeepDeletedMessages.x (arm64)…
==> Compiling src/Features/StoriesAndMessages/DisableStorySeen.x (arm64)…
==> Compiling src/Features/StoriesAndMessages/SeenButtons.x (arm64)…
==> Compiling src/Features/General/HideTrendingSearches.x (arm64)…
==> Compiling src/Features/General/CopyDescription.x (arm64)…
==> Compiling src/Features/General/NoRecentSearches.x (arm64)…
==> Compiling src/Tweak.x (arm64)…
==> Compiling src/Features/Confirm/CallConfirm.x (arm64)…
==> Compiling src/Features/Confirm/PostCommentConfirm.x (arm64)…
==> Compiling src/Features/Confirm/DMAudioMsgConfirm.x (arm64)…
==> Compiling src/Features/Confirm/StickerInteractConfirm.x (arm64)…
==> Compiling src/Features/General/BHSettingsMenuEntry.x (arm64)…
==> Compiling src/Features/Confirm/LikeConfirm.x (arm64)…
==> Compiling src/Features/Confirm/FollowConfirm.x (arm64)…
==> Compiling src/Features/Feed/HideSuggestedReels.x (arm64)…
==> Compiling src/Features/Feed/HideSuggestedForYou.x (arm64)…
==> Compiling src/Features/Feed/HideThreads.x (arm64)…
==> Compiling src/Features/Feed/HideAds.x (arm64)…
==> Compiling src/Features/Save/ProfileImageSave.x (arm64)…
==> Compiling src/SideloadFix.x (arm64)…
==> Compiling src/Features/Save/StoriesSave.x (arm64)…
==> Compiling src/Features/General/HideExploreGrid.xm (arm64)…
==> Compiling src/Features/Save/MediaSave.x (arm64)…
==> Linking tweak BHInsta (arm64)…
ld: warning: -multiply_defined is obsolete
ld: warning: ignoring duplicate libraries: '-lc++'
==> Generating debug symbols for BHInsta…
==> Merging tweak BHInsta…
==> Signing BHInsta…
Building the IPA.
jq: error (at <stdin>:1): Cannot iterate over null (null)
jq: error (at <stdin>:1): Cannot iterate over null (null)
[*] Couldn't Refresh 
[*] Couldn't find /Users/jthienel/Downloads/SCInsta/modules/libflex/.theos/obj/debug/libbhFLEX.dylib
[*] Couldn't find /var/folders/br/5kbrswh12m1_kqp3d1bsdk780000gn/T/tmp.5zxoyyyuFX/debs//Users/jthienel/Downloads/SCInsta/modules/libflex/.theos/obj/debug/libbhFLEX.dylib.deb
ar: /var/folders/br/5kbrswh12m1_kqp3d1bsdk780000gn/T/tmp.5zxoyyyuFX/debs//Users/jthienel/Downloads/SCInsta/modules/libflex/.theos/obj/debug/libbhFLEX.dylib.deb: No such file or directory
tar: Error opening archive: Failed to open 'data.tar.*'
[*] Couldn't extract libbhFLEX.dylib.deb
Done, we hope you enjoy BHInsta!

**Comment by SoCuul** [comment]
Ah I know exactly why it's not working. In version 0.3.0 the build script had an issue where it would not compile the FLEX tweak. This will be fixed in the upcoming version (working super hard on getting it out), but as a workaround try this before running the build script again.

```diff
In the file called "Makefile" replace the red text with the green text:
- SUBPROJECTS += libflex
+ BHInsta_SUBPROJECTS += modules/libflex
```

**Reply by Cody47** [reply]
**"Long press on the Instagram settings button to bring up the SCInsta tweak settings"**, doesn't works anymore on the latest version of the instagram v330.0.0!

**Reply by SoCuul** [reply]
> **"Long press on the Instagram settings button to bring up the SCInsta tweak settings"**, doesn't works anymore on the latest version of the instagram v330.0.0!

Yup, known issue. Will be fixed in the next release hopefully out this weekend 🤞

In the meantime just use an ipa before version v329.0.0

**Reply by InvalidPandaa** [reply]
Now it gives me this:

Building the IPA.
jq: error (at <stdin>:1): Cannot iterate over null (null)
jq: error (at <stdin>:1): Cannot iterate over null (null)
[*] Couldn't Refresh 
[*] Couldn't find /Users/jthienel/Downloads/SCInsta/modules/libflex/.theos/obj/debug/libbhFLEX.dylib
[*] Couldn't find /var/folders/br/5kbrswh12m1_kqp3d1bsdk780000gn/T/tmp.6giiCFPcTo/debs//Users/jthienel/Downloads/SCInsta/modules/libflex/.theos/obj/debug/libbhFLEX.dylib.deb
ar: /var/folders/br/5kbrswh12m1_kqp3d1bsdk780000gn/T/tmp.6giiCFPcTo/debs//Users/jthienel/Downloads/SCInsta/modules/libflex/.theos/obj/debug/libbhFLEX.dylib.deb: No such file or directory
tar: Error opening archive: Failed to open 'data.tar.*'
[*] Couldn't extract libbhFLEX.dylib.deb
Done, we hope you enjoy BHInsta!

**Reply by InvalidPandaa** [reply]
Ig it's the same error

**Reply by Cody47** [reply]
[libbhFLEX.zip](https://github.com/SoCuul/SCInsta/files/15283753/libbhFLEX.zip)
bruh, just extract and copy the dylib file manually in the directory, its a temp fix but works!

**Comment by Cody47** [comment]
> **"Long press on the Instagram settings button to bring up the SCInsta tweak settings"**, doesn't works anymore on the latest version of the instagram v330.0.0!

@SoCuul any updates on this?

**Reply by SoCuul** [reply]
https://github.com/SoCuul/SCInsta/releases/tag/v0.4.0 🎉

**Comment by SoCuul** [comment, marked as answer]
@InvalidPandaa Should be completely fixed in the newest version.
https://github.com/SoCuul/SCInsta/releases/tag/v0.4.0

**Reply by InvalidPandaa** [reply]
Thank you, it worked! PS: you may could add a note that the user needs to manually create the "packages" folder when using sideload

**Reply by SoCuul** [reply]
Actually, I accidentally messed up the gitignore as the folder should be uploaded. Thanks for telling me though, I had no idea haha.

**Reply by InvalidPandaa** [reply]
You're welcome!
